modules/fedimint-ln-client/src/receive.rs

```diff
@@ -0,0 +1,385 @@
+        let timeout = Duration::from_secs(15);
+        global_context
+            .api()
+            .await_output_outcome::<LightningOutputOutcome>(outpoint, timeout, &module_decoder)
+            .await
+            .map_err(|_| LightningReceiveError::Rejected)?;
+        let confirmed_invoice = ConfirmedInvoice {
+            invoice,
+            keypair: payment_keypair,
+        };
+
+        Ok(confirmed_invoice)
+    }
+
+    async fn await_invoice_rejection(
+        global_context: DynGlobalClientContext,
+        common: LightningReceiveCommon,
+        txid: TransactionId,
+    ) {
+        global_context
+            .await_tx_rejected(common.operation_id, txid)
+            .await;
+    }
+
+    async fn transition_confirmed_invoice(
+        result: Result<ConfirmedInvoice, LightningReceiveError>,
+        old_state: LightningReceiveStateMachine,
+    ) -> LightningReceiveStateMachine {
+        match result {
+            Ok(confirmed_invoice) => LightningReceiveStateMachine {
+                common: old_state.common,
+                state: LightningReceiveStates::ConfirmedInvoice(LightningReceiveConfirmedInvoice {
+                    confirmed_invoice,
+                }),
+            },
+            Err(e) => LightningReceiveStateMachine {
+                common: old_state.common,
+                state: LightningReceiveStates::Canceled(e),
+            },
+        }
+    }
+
+    async fn transition_invoice_rejected(
+        old_state: LightningReceiveStateMachine,
+    ) -> LightningReceiveStateMachine {
+        LightningReceiveStateMachine {
+            common: old_state.common,
+            state: LightningReceiveStates::Canceled(LightningReceiveError::Rejected),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Decodable, Encodable)]
+pub struct LightningReceiveConfirmedInvoice {
+    confirmed_invoice: ConfirmedInvoice,
+}
+
+impl LightningReceiveConfirmedInvoice {
+    fn transitions(
+        &self,
+        global_context: &DynGlobalClientContext,
+    ) -> Vec<StateTransition<LightningReceiveStateMachine>> {
+        let invoice = self.confirmed_invoice.invoice.clone();
+        let keypair = self.confirmed_invoice.keypair;
+        let global_context = global_context.clone();
+        vec![
+            StateTransition::new(
+                Self::await_incoming_contract_account(invoice, global_context.clone()),
+                move |dbtx, contract, old_state| {
+                    Box::pin(Self::transition_funded(
+                        old_state,
+                        keypair,
+                        contract,
+                        dbtx,
+                        global_context.clone(),
+                    ))
+                },
+            ),
+            // TODO: Do we even want this case?
+            // Should it just be set to the invoice expiry?
+            StateTransition::new(Self::await_payment_timeout(), |_dbtx, (), old_state| {
```

Comment:
I think the timeout here should always be set to the invoice expiry? Otherwise its possible for the invoice to be paid and the contract funded but the client SM won't be waiting to claim it.